polynote-frontend/polynote/ui/component/notebook/cell.ts

```diff
@@ -786,6 +786,15 @@ abstract class MonacoCell extends Cell {
             // overflowWidgetsDomNode: this.overflowDomNode
         });
 
+        // None of the official ways to do this seem to work
+
```

Comment:
This is really unfortunate, but it seems to be the "correct-est" way to get sane behavior out of jump-to-definition. Monaco still calls to the server as soon as you hover a symbol, but it won't try to open a tab until you click (before, any time monaco initiated a jump-to-definition, it would result in polynote going forward with it. The monaco update included a change where monaco initiates a jump-to-definition on cmd-hover, for a "peek" feature that we're not using but can't be turned off)

Reply:
Trying to understand how this works. I guess it's monkey-patching [this function](https://github.com/microsoft/vscode/blob/dd112ec0243c4b42bb3106e64df1688e242a6559/src/vs/editor/browser/services/abstractCodeEditorService.ts#L275) ?

So basically we have Monaco ignore all "codeEditorOpenHandlers", whatever that is? 

> Monaco still calls to the server as soon as you hover a symbol

Are we just ignoring those calls then? Is there any way we can just swallow mouse-hover-while-holding-down-cmd events before they get to Monaco or is that too hacky?

Reply:
Yeah, it's monkey patching. Supposedly there's some way to override this stuff from the editor constructor, but the examples are from years ago and don't work anymore.

> or is that too hacky?

... I mean. Look at this code. But I think this is a little closer to working with Monaco than against. And theoretically we could support the peek feature now (because Monaco gets a real http url that it can fetch the definition source code from)

Reply:
> Look at this code. 
😅 

> Monaco gets a real http url that it can fetch the definition source code from

Ah, I see. Monaco needs to be able to GET the URI independently in order to implement the "peek" feature? Too bad it can't just ask for data like it does for completions, etc. Have you considered monkey-patching that part or do you feel like it'd be too big a change to make (working against Monaco too much).